partner_email_check/models/res_partner.py

```diff
@@ -0,0 +1,101 @@
+
+import logging
+
+from odoo import _, api, models
+from odoo.exceptions import UserError, ValidationError
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from email_validator import (
+        EmailSyntaxError,
+        EmailUndeliverableError,
+        validate_email,
+    )
+except ImportError:
+    _logger.debug('Cannot import "email_validator".')
+
+    validate_email = None
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def copy_data(self, default=None):
+        res = super(ResPartner, self).copy_data(default=default)
+        if self._should_filter_duplicates():
+            for copy_vals in res:
+                copy_vals.pop("email", None)
+        return res
+
+    @api.model
+    def email_check(self, emails):
+        return ",".join(
+            self._normalize_email(email.strip())
+            for email in emails.split(",")
+            if email.strip()
+        )
+
+    @api.constrains("email")
+    def _check_email_unique(self):
+        if self._should_filter_duplicates():
+            for rec in self.filtered("email"):
+                if "," in rec.email:
+                    raise UserError(
+                        _(
+                            "Field contains multiple email addresses. This is "
+                            "not supported when duplicate email addresses are "
+                            "not allowed."
+                        )
+                    )
+                if self.search_count([("email", "=", rec.email), ("id", "!=", rec.id)]):
+                    raise UserError(
+                        _("Email '%s' is already in use.") % rec.email.strip()
+                    )
+
+    def _normalize_email(self, email):
+        if not self._should_check_syntax():
+            return email
+        if validate_email is None:
+            _logger.warning(
+                "Can not validate email, "
+                'python dependency required "email_validator"'
+            )
+            return email
+
+        try:
+            result = validate_email(
+                email,
+                check_deliverability=self._should_check_deliverability(),
+            )
+        except EmailSyntaxError:
+            raise ValidationError(
+                _("%s is an invalid email") % email.strip()
+            ) from EmailSyntaxError
+        except EmailUndeliverableError:
+            raise ValidationError(
+                _("Cannot deliver to email address %s") % email.strip()
+            ) from EmailUndeliverableError
+        return result["local"].lower() + "@" + result["domain_i18n"]
+
+    def _should_check_syntax(self):
+        return self.env.company.partner_email_check_syntax
+
+    def _should_filter_duplicates(self):
+        return self.env.company.partner_email_check_filter_duplicates
+
+    def _should_check_deliverability(self):
+        return self.env.company.partner_email_check_check_deliverability
+
+    @api.model
+    def create(self, vals):
```

Comment:
Please override the create method in batch (using `api.model_create_multi` decorator).

Reply:
Hi @diogocsc 
This won't work. With this decorator the method now takes a list of dicts as argument.
See https://github.com/odoo/odoo/blob/3ed088ae93bdfe5565205949779cae8b7c1d34e7/odoo/models.py#L3847

Reply:
Hi @marielejeune you are correct. I noticed when tests failed, and already added a fix.
Please check again.

Reply:
Yes, it is now correct. I think it would be better to fixup your commit inside the previous one (migration commit), for history purposes.

Reply:
@marielejeune any suggestion on how to apply the fixup? I'm still a bit rookie in what concerns using git rebase -i... I tried to do just that but only my last commit showed up...

Reply:
Hi, no problem. You have to provide the sha of the first commit you don't want to modify, so here it would be 0294050bcac171c619ba2e34dc5c487f4eb0fd50.
Then you don't modify the migration commit but you fixup the other 2 commits inside it (with "f" shortcut in the interactive rebase menu)

Reply:
Hi @marielejeune , I believe the commit to fixup into is 0a01cc78634a4603ae1ee8f6713117e0207720fb, as the one you mentioned still add the 15 version tag... correct?

Reply:
oh...I just noticed it only shows the commits after the one I mentioned...will do as you say then

Reply:
Yeah, sorry if it wasn't completely clear :)

Reply:
Done in my last commit. Please check